language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/EnumClass.scala

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.codegen.backend.java.inner
+
+import com.daml.ledger.javaapi
+import com.digitalasset.daml.lf.data.Ref.Identifier
+import com.digitalasset.daml.lf.iface
+import com.squareup.javapoet._
+import com.typesafe.scalalogging.StrictLogging
+import javax.lang.model.element.Modifier
+
+private[inner] object EnumClass extends StrictLogging {
+
+  def generate(
+      className: ClassName,
+      identifier: Identifier,
+      enum: iface.Enum,
+  ): TypeSpec = {
+
+    logger.info("Start")
+    val enumType = TypeSpec.enumBuilder(className).addModifiers(Modifier.PUBLIC)
+    enum.constructors.foreach(c => enumType.addEnumConstant(c.toUpperCase()))
+    enumType.addField(generateValuesArray(enum))
+    enumType.addMethod(generateEnumsMapBuilder(enum))
+    enumType.addField(generateEnumsMap(className))
+    enumType.addMethod(generateFromValue(className, enum))
+    enumType.addMethod(generateToValue(className))
+    logger.debug("End")
+    enumType.build()
+  }
+
+  private def generateValuesArray(enum: iface.Enum): FieldSpec =
+    FieldSpec
+      .builder(ArrayTypeName.of(classOf[javaapi.data.DamlEnum]), "__values$")
+      .addModifiers(Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+      .initializer(CodeBlock.of(
```

Comment:
I think it would be more to the style of JavaPoet to initialize the field `__values$` like this:
```scala
val constructorValues = enum.constructors.map(c => CodeBlock.of("new $T($S))", classOf[javaapi.data.DamlEnum], c)).toIterable.asJava
FieldSpec.initializer(CodeBlock.join(constructorValues, ", "))
```

Reply:
fixed

Reply:
fixed

Reply:
Ah right, because it's an array initializer. Thanks 👍